src/DotNetCore.CAP.RedisStreams/CapOptions.Redis.cs

```diff
@@ -2,18 +2,40 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using StackExchange.Redis;
+using System.Linq;
+using System.Net;
 
 // ReSharper disable once CheckNamespace
 namespace DotNetCore.CAP
 {
     public class CapRedisOptions
     {
+        private ConfigurationOptions? _configuration;
+
         /// <summary>
         ///     Gets or sets the native options of StackExchange.Redis
         /// </summary>
-        public ConfigurationOptions? Configuration { get; set; }
+        public ConfigurationOptions? Configuration
+        {
+            get => _configuration;
+            set
+            {
+                _configuration = value;
+
+                Endpoint = _configuration == null
```

Comment:
Why do you add the code?

Reply:
Because there's no value assigned to `Endpoint` anywhere and cause NRE.

Reply:
I think endpoint is the target transport server hosting, if it is null that CAP will not work

Reply:
If you think that's what you want, remove the `!` and fix the compiler error.

Reply:
But Configuration allows a null value to be assigned.